lib/sender.py

```diff
@@ -15,39 +15,15 @@ def __init__(self, queue, riemann_client, enable_threads):
         self.enable_threads = enable_threads
         self.daemon = True
 
-    def send(self, events):
-        if len(events) > 0:
-            log.debug("Sending %s events..." % (len(events)))
-            while len(events) > 0:
-                event = events.pop(0)
-                try:
-                    log.debug("Event: %s" % (event.dict()))
-                    self.riemann.send(event.dict())
-                except socket.error:
-                    log.error("Unable to send event '%s' to %s:%s" % (event.service, self.riemann.host, self.riemann.port))
-        else:
-            log.warning("Send called with no events to send.")
-
     def run(self):
         while self.enable_threads:
-            log.debug("EventSender %s: waiting for a task..." % (self.name))
-            task = self.queue.get(block=True)
-
-            if task == "exit":
-                log.debug("%s: received 'exit' event" % (self.name))
-                break
+            log.debug("EventSender %s: waiting for an event..." % (self.name))
+            event = self.queue.get(block=True)
 
             try:
-                log.debug("%s: Waiting for events from '%s'" % (self.name, task.name))
-
-                events = task.drain()
-
-                log.debug("%s: Waiting complete - attempting to send events - %s" % (self.name, task.name))
-
-                self.send(events)
-            except Exception as e:
-                log.error("Exception sending events from '%s': %s" % (task.name, str(e)))
+                log.debug("Sending event: %s" % (event.dict()))
+                self.riemann.send(event.dict())
+            except socket.error:
```

Comment:
Do we need to handle any re-initialization here?

Reply:
Does not look like it:

https://github.com/banjiewen/bernhard/blob/master/bernhard/__init__.py#L236-L242

I think we are OK from an error checking / reconnecting standpoint based on looking at bernhard. We could potentially add retry logic, for example, but we did not have this w/ UDP.